From cdc5ffb91f0bf80bdf1281f4e0132b225a3ee0bc Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Fri, 16 Jan 2015 21:34:36 -0800
Subject: [PATCH 3003/3005] Fix <local> and <iterator> when building with Lion
 and newer Availability.h

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit cb271d6ce269bbddc916643c495ccbdfde6e3054)
---
 include/iterator | 4 ++--
 include/locale   | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git llvm_release_35/projects/libcxx/include/iterator macports_release_35/projects/libcxx/include/iterator
index f338e01..64b52e7 100644
--- llvm_release_35/projects/libcxx/include/iterator
+++ macports_release_35/projects/libcxx/include/iterator
@@ -913,8 +913,8 @@ public:
     _LIBCPP_INLINE_VISIBILITY bool failed() const _NOEXCEPT {return __sbuf_ == 0;}
 
 #if !defined(__APPLE__) || \
-    (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED > __MAC_10_8) || \
-    (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED > __IPHONE_6_0)
+    (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && defined(__MAC_10_8) && __MAC_OS_X_VERSION_MIN_REQUIRED > __MAC_10_8) || \
+    (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && defined(__IPHONE_6_0) && __IPHONE_OS_VERSION_MIN_REQUIRED > __IPHONE_6_0)
 
     template <class _Ch, class _Tr>
     friend
diff --git llvm_release_35/projects/libcxx/include/locale macports_release_35/projects/libcxx/include/locale
index fcff402..10d8ca9 100644
--- llvm_release_35/projects/libcxx/include/locale
+++ macports_release_35/projects/libcxx/include/locale
@@ -1467,8 +1467,8 @@ __pad_and_output(_OutputIterator __s,
 }
 
 #if !defined(__APPLE__) || \
-    (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED > __MAC_10_8) || \
-    (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED > __IPHONE_6_0)
+    (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && defined(__MAC_10_8) && __MAC_OS_X_VERSION_MIN_REQUIRED > __MAC_10_8) || \
+    (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && defined(__IPHONE_6_0) && __IPHONE_OS_VERSION_MIN_REQUIRED > __IPHONE_6_0)
 
 template <class _CharT, class _Traits>
 _LIBCPP_HIDDEN
-- 
2.2.2

